src/libs/ttf/lv_ttf.c

```diff
@@ -0,0 +1,4512 @@
+    else if (b0 == 28)
+        return stbtt__buf_get16(b);
+    else if (b0 == 29)
+        return stbtt__buf_get32(b);
+    STBTT_assert(0);
+    return 0;
+}
+
+static void stbtt__cff_skip_operand(stbtt__buf *b) {
+    int v, b0 = stbtt__buf_peek8(b);
+    STBTT_assert(b0 >= 28);
+    if (b0 == 30) {
+        stbtt__buf_skip(b, 1);
+        while (b->cursor < b->size) {
+            v = stbtt__buf_get8(b);
+            if ((v & 0xF) == 0xF || (v >> 4) == 0xF)
+                break;
+        }
+    } else {
+        stbtt__cff_int(b);
+    }
+}
+
+static stbtt__buf stbtt__dict_get(stbtt__buf *b, int key) {
+    stbtt__buf_seek(b, 0);
+    while (b->cursor < b->size) {
+        int start = b->cursor, end, op;
+        while (stbtt__buf_peek8(b) >= 28)
+            stbtt__cff_skip_operand(b);
+        end = b->cursor;
+        op = stbtt__buf_get8(b);
+        if (op == 12)
+            op = stbtt__buf_get8(b) | 0x100;
+        if (op == key)
+            return stbtt__buf_range(b, start, end - start);
+    }
+    return stbtt__buf_range(b, 0, 0);
+}
+
+static void stbtt__dict_get_ints(stbtt__buf *b, int key, int outcount, stbtt_uint32 *out) {
+    int i;
+    stbtt__buf operands = stbtt__dict_get(b, key);
+    for (i = 0; i < outcount && operands.cursor < operands.size; i++)
+        out[i] = stbtt__cff_int(&operands);
+}
+
+static int stbtt__cff_index_count(stbtt__buf *b) {
+    stbtt__buf_seek(b, 0);
+    return stbtt__buf_get16(b);
+}
+
+static stbtt__buf stbtt__cff_index_get(stbtt__buf b, int i) {
+    int count, offsize, start, end;
+    stbtt__buf_seek(&b, 0);
+    count = stbtt__buf_get16(&b);
+    offsize = stbtt__buf_get8(&b);
+    STBTT_assert(i >= 0 && i < count);
+    STBTT_assert(offsize >= 1 && offsize <= 4);
+    stbtt__buf_skip(&b, i * offsize);
+    start = stbtt__buf_get(&b, offsize);
+    end = stbtt__buf_get(&b, offsize);
+    return stbtt__buf_range(&b, 2 + (count + 1) * offsize + start, end - start);
+}
+
+//////////////////////////////////////////////////////////////////////////
+//
+// accessors to parse data from file
+//
+
+// on platforms that don't allow misaligned reads, if we want to allow
+// truetype fonts that aren't padded to alignment, define ALLOW_UNALIGNED_TRUETYPE
+
+#define ttBYTE(p) (*(stbtt_uint8 *)(p))
+#define ttCHAR(p) (*(stbtt_int8 *)(p))
+#define ttFixed(p) ttLONG(p)
+
+/*static stbtt_uint16 ttUSHORT(stbtt_uint8 *p)
+{
+   return p[0] * 256 + p[1];
+}*/
+// static stbtt_int16 ttSHORT(stbtt_uint8 *p) { return p[0] * 256 + p[1]; }
+// static stbtt_uint32 ttULONG(stbtt_uint8 *p) { return (p[0] << 24) + (p[1] << 16) + (p[2] << 8) + p[3]; }
+// static stbtt_int32 ttLONG(stbtt_uint8 *p) { return (p[0] << 24) + (p[1] << 16) + (p[2] << 8) + p[3]; }
+
+#define stbtt_tag4(p, c0, c1, c2, c3) ((p)[0] == (c0) && (p)[1] == (c1) && (p)[2] == (c2) && (p)[3] == (c3))
+#define stbtt_tag(p, str) stbtt_tag4(p, str[0], str[1], str[2], str[3])
+
+/*static int stbtt__isfont(stbtt_uint8 *font)
```

Comment:
Can we remove the many commented out code?

Reply:
absolutely. The reason I left it in, honestly, is because I've removed support for certain formats and rendering modes to save flash space, and I wasn't sure how many font files that would impact. I left that code in there so I could put like, OS/2 TTF and SVG support back in if necessary. It's fine to remove it. I thought about doing it, but I've been on fence about adding this stuff back in.